app/services/market_analysis_core.py

```
@@ -2146,15 +2145,207 @@ async def monitor_spreads(
+        """Discover real spot assets from exchange APIs."""
+        try:
+            if exchange.lower() == "binance":
+                return await self._discover_binance_assets()
+            elif exchange.lower() == "kraken":
+                return await self._discover_kraken_assets()
+            elif exchange.lower() == "kucoin":
+                return await self._discover_kucoin_assets()
+            else:
+                return None
+                
+        except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)